### ISSUE TITLE
After updating from 5.x to latest 6.30, KFImage lost auto size before fully load the image, it was good with 5.x

### ISSUE THREAD

**Issue by iaomw** [onevcat/Kingfisher#1721]
### Check List

Thanks for considering to open an issue. Before you submit your issue, please confirm these boxes are checked.

- [x] I have read the [wiki page](https://github.com/onevcat/Kingfisher/wiki) and [cheat sheet](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet), but there is no information I need.
- [x] I have searched in [existing issues](https://github.com/onevcat/Kingfisher/issues?utf8=✓&q=is%3Aissue), but did not find a same one.
- [x] I want to report a problem instead of asking a question. It'd better to use [kingfisher tag in Stack Overflow](http://stackoverflow.com/questions/tagged/kingfisher) to ask a question.

### Issue Description
Update Kingfisher 5.x to 6.30, then my UI got totally messed up. Previously, `KFImage` has some kind of auto size from parent view, and the aspect ratio of KFImage looks like 1:1. Well, that's fine, everything just work out of box.

Sadly with 6.3.0, it didn't fill up the parent view. The `KFImage` shrink itself to very little size, probably the size of placeholder view. After `KFImage` got the image fully loaded, it will grow up immediately without animation and sparking my eye. In the same manner, my SwiftUI List will jumping here and there like rabbit.
![image](https://user-images.githubusercontent.com/2307692/120080966-a56b8000-c0ed-11eb-81f9-a070dc96d406.png)
 
#### What

```
ScrollViewReader { proxy in
                
                List {
                    ForEach(idx, id: \.self) { index in
                        
                        let tmp = index * col
                        
                        HStack(alignment: .center) {
                            ForEach(0..<col, id: \.self) { offset in

                                let preview_link = ...

                                Button(action: {}, label: { 
                                    KFImage.url(URL(string: preview_link))
                                        .scaleFactor(scale)
                                        .backgroundDecode()
                                        .fade(duration: 0.2)
                                        .cancelOnDisappear(true)
                                        .placeholder {ProgressView()}
                                        .resizable().scaledToFit()
```

#### Other Comment
I tried using `GeometryReader` to give a size for `KFImage`, but didn't get good result. That's not feasible.
Rollback to 5.x is possible, but 6.3 has visible better performance with SwiftUI.
Are there any little parameter to make it work as 5.x sizing ?

**Comment by onevcat** [comment]
Not sure what is going on here, but from your code, it seems that you didn't set any frame to `KFImage` (or its parent cell) so it just takes what is it needs, which is the placeholder here.

To allow it fitting with its parent, maybe you can add a `.frame(width: nil, height: nil)` or something similar. Or, if you know an estimated size, you can also give it to your `ProgressView` so the transition from placeholder to the actual image would looks better.

**Comment by iaomw** [comment]
Well, I didn't give it frame size, because previously it just work without any problem.

Then, I tested some older version, 6.0.0 and 6.0.1 works fine without this issue. 
For the same code, it behaves the same as 5.x but has better performance. 
If there is no big issue, I would stay at 6.0.1 for a long time.
But it's good to know what is broken after 6.1.0.

![image](https://user-images.githubusercontent.com/2307692/120097574-c5d62180-c163-11eb-92c1-e364d2b446be.png)

**Comment by iaomw** [comment]
The bug has been fixed with 7.0.0.

**Comment by sochalewski** [comment]
@onevcat It seems that this problem occurs again in 7.3.1, but 7.0.0-7.3.0 work fine.

**Comment by onevcat** [comment]
Ummm.. @sochalewski Thanks for the feedback. So it seems [this causes](https://github.com/onevcat/Kingfisher/issues/1972) it. Let me see what to do...

**Comment by onevcat** [comment]
Fixed in #1975 and I will tag it soon.
